src/pages/index.js

```diff
@@ -85,62 +121,123 @@ const profileModal = new PopupWithForm(
   editValidator
 );
 
-const addModal = new PopupWithForm(
+const profileAvatar = new PopupWithForm(
   {
-    popupSelector: "#add-card-modal",
-    handleFormSubmit: handleAddCardFormSubmit,
+    popupSelector: "#profile-avatar-modal",
+    handleFormSubmit: handleAvatarSubmit,
   },
-  cardAddValidator
+  avatarValidator
 );
 
-const popupImage = new PopupWithImage("#preview-modal");
+const confirmDeletePopup = new PopupConfirmDelete({
+  popupSelector: "#delete-card-modal",
+  handleFormSubmit: () => {},
+});
 
-const section = new Section(
+const addModal = new PopupWithForm(
   {
-    items: initialCards,
-    renderer: renderCard,
+    popupSelector: "#add-card-modal",
+    handleFormSubmit: handleAddCardFormSubmit,
   },
-  ".cards__list"
-);
-
+  cardAddValidator
+); //
+
+/*                        PopupWithImage                      */
+const popupImage = new PopupWithImage("#preview-modal"); //
+
+//initial cards from live overview
+/*
+api
+  .getInitialCards()
+  .then((cards) => {
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(err);
+  }); */
+
+/*                        UserInfo                       */
 const userInfo = new UserInfo({
   nameElement: ".profile__title",
   jobElement: ".profile__description",
+  avatarElement: ".profile__image",
 });
 
+////refactor under handle profile edit and apis
+/*
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    description: userData.about,
+  });
+}); */
+
+/*                        component EventListeners                      */
 profileModal.setEventListeners();
 
 popupImage.setEventListeners();
-section.renderItems();
+
+addModal.setEventListeners();
+
+confirmDeletePopup.setEventListeners();
+
+profileAvatar.setEventListeners();
 
 /* -------------------------------------------------------------------------- */
 /*                        functions                      */
 /* -------------------------------------------------------------------------- */
 
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleLikeIcon,
+    handleDeleteCard
+  );
+  return card.getView();
+}
+
 function renderCard(item, method = "addItem") {
   const cardElement = createCard(item);
   section[method](cardElement);
 }
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
-}
-
-///establish handleImageClick
+///establish handleImageClick///
 function handleImageClick(data) {
   popupImage.open({ name: data.name, link: data.link });
 }
 
 //// use userInfo for handlesubmit
+/*
 function handleProfileEditSubmit(inputValue) {
   userInfo.setUserInfo({
     name: inputValue.title,
     description: inputValue.description,
   });
   profileModal.close();
+} */
+
+function handleProfileEditSubmit(userData) {
+  profileModal.renderLoad(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```